caddyhttp/proxy/body.go

```diff
@@ -23,9 +30,19 @@ func (b *bufferedBody) rewind() error {
 	return err
 }
 
+type unbufferedBody struct {
+	io.ReadCloser
+}
+
+var _ rewindableReader = &unbufferedBody{}
+
+func (b *unbufferedBody) rewind() error {
+	panic("cannot rewind unbuffered body")
```

Comment:
Ah yes... When I wrote this method it wasn't hidden behind a common interface. Calling `rewind()` on it was thus basically a precondition failure, akin to how `501 Not Implemented` is a server error code instead of being a client error like 404. I still kind of think that it's a precondition failure, but since this struct is only used using the `rewindableReader` it's surely not the right choice anymore. I'll change it.